File: Top_Interview_Questions/14_LongestCommonPrefix.cpp
class Solution {
public:
    string longestCommonPrefix(vector<string>& strs) {
        if(strs.empty()) return "";
        int cur = 0;
        string ans = "";
        while(1) {
            char last;
            if(cur >= strs[0].size())
                return ans;
            last = strs[0][cur];
            for(int i = 1; i < strs.size(); ++i) {           
                if(cur >= strs[i].size())
                    return ans;
                if(last != strs[i][cur])
                    return ans;
            }
            ans += last;
            ++cur;
        }
        return ans;
    }
};

File: Question_Go/14_Longest_Common_Prefix.go
func longestCommonPrefix(strs []string) string {
    if len(strs) == 0 {
        return ""
    }
    
    var ans string
    for i := 0; i < len(strs[0]); i++ {
        ch := strs[0][i]
        for j := 1; j < len(strs); j++ {
            if i >= len(strs[j]) || ch != strs[j][i] {
                return ans
            }
        }
        ans += string(ch)
    }
    return ans
}